.ci/packaging.groovy

```diff
@@ -35,56 +36,49 @@ pipeline {
       }
     }
     stage('Linux'){
-      agent { label 'ubuntu && immutable' }
       options { skipDefaultCheckout() }
-      when {
-        beforeAgent true
-        expression { return params.linux }
-      }
-      environment {
-        SNAPSHOT = "true"
-        PLATFORMS = "+linux/armv7 +linux/ppc64le +linux/s390x +linux/mips64"
-        HOME = "${env.WORKSPACE}"
-      }
-      steps {
-        deleteDir()
-        unstash 'source'
-        dir("${BASE_DIR}"){
-          sh(label: 'Release', script: './dev-tools/jenkins_release.sh')
+      matrix {
+        agent { label 'ubuntu && immutable' }
+        axes {
+          axis {
+            name 'PLATFORMS'
+            values '+linux/armv7', '+linux/ppc64le', '+linux/s390x', '+linux/mips64', 'darwin'
+          }
         }
-        publishPackages()
-      }
-    }
-    stage('Mac OS'){
-      agent { label 'macosx' }
-      options { skipDefaultCheckout() }
-      when {
-        beforeAgent true
-        expression { return params.linux }
-      }
-      environment {
-        KEYCHAIN = "/var/lib/jenkins/Library/Keychains/Elastic.keychain-db"
-        SNAPSHOT = "true"
-        PLATFORMS = "darwin"
-        APPLE_SIGNING_ENABLED = "true"
-        HOME = "${env.WORKSPACE}"
-      }
-      steps {
-        deleteDir()
-        unstash 'source'
-        withMaskEnv( vars: [
-            [var: "KEYCHAIN_PASS", password: getVaultSecret(secret: "secret/jenkins-ci/macos-codesign-keychain").data.password],
-        ]){
-          dir("${BASE_DIR}"){
-            sh(label: 'Release', script: './dev-tools/jenkins_release.sh')
+        stages {
+          stage('Package'){
+            environment {
+              HOME = "${env.WORKSPACE}"
+            }
+            steps {
+              deleteDir()
+              unstash 'source'
+              release()
+              publishPackages()
+            }
           }
-          publishPackages()
         }
       }
     }
   }
 }
 
+def release(){
+  dir("${BASE_DIR}"){
+    if(env.PLATFORMS == 'darwin' && params.macos){
+      withMaskEnv( vars: [
+          [var: "KEYCHAIN_PASS", password: getVaultSecret(secret: "secret/jenkins-ci/macos-codesign-keychain").data.password],
+          [var: "KEYCHAIN", password: "/var/lib/jenkins/Library/Keychains/Elastic.keychain-db"],
+          [var: "APPLE_SIGNING_ENABLED", password: "true"],
+      ]){
+        sh(label: 'Release', script: './dev-tools/jenkins_release.sh')
+      }
+    } else if (params.linux){
+      sh(label: 'Release', script: './dev-tools/jenkins_release.sh')
```

Comment:
Related but unrelated, what do you think to change `./dev-tools/jenkins_release.sh` to call `make release-manager-snapshot` and remove the SNAPSHOT reference in the script and also in this packaging.groovy file?

See https://github.com/elastic/beats/blob/65b52553d69f5ff79f9a6eec39b6bdea9f92c8c9/Makefile#L185-L186

Reply:
but in a follow-up, it requires more changes it is nor as easy as changing the command

Reply:
we change to use `mage package` so the change is not needed